### ISSUE TITLE
Unclear how to make the connection idle

### ISSUE THREAD

**Issue by mjonss** [vlet/p5-Protocol-HTTP2#1]
How is one suppose to use this to add new requests sequentially?

Since it will finish the connection (by sending an GOAWAY) as soon as there are no active_sessions.

I want to be able to create a new request and then wait for the reply and then create another request etc. like:

``` perl
while (my $msg = get_new_msg()) {
    $client->request(... $msg ..., on_done => ... );
    while (my $frame = $client->next_frame) {
        syswrite $socket, $frame;
    }
    my $reply;
    sysread $socket, $reply, 4096;
    $client->feed($reply);
    # Now there will be no active streams!
}
```

Currently it will finish the connection in Client::active_streams
223     $self->{con}->finish unless $self->{active_streams} > 0;
which prohibits subsequent use of the http2 connection.

As a workaround I can do $client->active_streams(+1) when I have created $client.

I propose $client->allow_idle(0|1) to control this instead and perhaps a defined way to shutdown it down like $client->shutdown.

**Comment by vlet** [comment]
Method/parameter `keepalive` sounds more familiar and intuitive for this feature.
Unfortunately `shutdown` already taken as a getter for connection status, so probably `close` should be implemented instead

**Comment by vlet** [comment]
fixed in 1.06
